zombies.js

```diff
@@ -208,6 +237,136 @@
  * @return {string/boolean}   Weapon name or false if nothing is equipped.
  */
 
+// class Player {
+//   constructor (name, health, strength, speed){
+//     this.name = name;
+//     this.health = health;
+//     this.strength = strength;
+//     this.speed = speed;
+//     this.isAlive = true;
+//     this.equipped = false;
+//     this._pack = [];
+//     this._maxHealth = health;
+//   }
+
+function Player (name, health, strength, speed){
+  this.name = name;
+  this.health = health;
+  this.strength = strength;
+  this.speed = speed;
+  this.isAlive = true;
+  this.equipped = false;
+  let pack = [];
+  this.getPack = function (){
+    return pack;
+  };
+  this.getMaxHealth = function (){
+    return health;
+  };
+}
+
+Player.prototype.checkPack = function (){
+  var packArr = [];
+  for (var i = 0; i < this.getPack().length; i++){
```

Comment:
It looks like you're creating a new array of pack items, just to join them together for the console.log below. This isn't really necessary since .join is not a mutating method.